### PR TITLE
ch-image: add --force support for fedora, install epel via url

### PR DESCRIPTION
**Pull request by heasterday**
This addresses #1026 and #1030.

Marking draft pending further local testing.

### CODE REVIEW

File: lib/fakeroot.py [hunk 1]
```diff
@@ -147,15 +147,18 @@
    # 1. CentOS seems to have only fakeroot, which is in EPEL, not the standard
    #    repos.
    #
-   # 2. Enabling EPEL can have undesirable side effects, e.g. different
+   # 2. Unlike on CentOS, RHEL doesn't have the epel-release rpm in the
+   #    standard repos, so install the rpm directly.
```
Comment (reidpr) [inline lib/fakeroot.py:151]: Should clarify we're doing the RPM way for both. Or did you mean “unlike on Fedora”?

File: lib/fakeroot.py [hunk 2]
```diff
@@ -221,6 +224,24 @@
...
+   #    release versions available for building (what was on Dockerhub). To
+   #    extend support to additional versions one need only adjust the match
+   #    config.
+   #
+   # 2. The fakeroot package is in the base repository set so enabling EPEL is
+   #    not required.
+
+   "fedora":
+   { "name": "Fedora 24-35,",
+     "match":  ("/etc/fedora-release", r"release (2[4-9]|3[0-5])"),
```
Comment (reidpr) [inline lib/fakeroot.py:239]: I think Dave Love had a good point. Can we remove the high end of this range test?

File: test/build/50_fakeroot.bats [hunk 3]
```diff
@@ -693,3 +709,191 @@ EOF
     [[ $output = *'will use --force'* ]]
     [[ $output = *'--force: init OK & modified 1 RUN instructions'* ]]
 }
+
+# We would prefer to test the lowest supported --force version, 24, but the
+# ancient version of dnf it has doesn't fail the transaction when a package
+# fails so we test with 26 instead.
```
Comment (reidpr) [inline test/build/50_fakeroot.bats:715]: Move this comment inside the test.

File: test/build/50_fakeroot.bats [hunk 4]
```diff
@@ -693,3 +709,191 @@ EOF
...
+    [[ $output = *'will use --force'* ]]
+    [[ $output = *'--force: init OK & modified 1 RUN instructions'* ]]
+}
+
+@test "${tag}: Fedora 35: unneeded, no --force, build succeeds" {
+    scope standard
+    # no commands that may need it, without --force, build succeeds
+    # also: correct config
+    run ch-image -v build -t fakeroot-temp -f - . <<'EOF'
+FROM fedora:35
```
Comment (reidpr) [inline test/build/50_fakeroot.bats:814]: Similar to above, can we do `fedora:latest` here? (which seems to be Fedora 33 currently) Goal being to test the latest released Fedora, not a pre-release, though? Or maybe we really do want `fedora:rawhide` so we can fix problems early?

**Review by reidpr — CHANGES_REQUESTED** [review]
Good work. Some comments inline.

Were you planning to add the Debian 11 (#1043) and openSUSE (#1044) from PR #1035 here?

See also #1042.

File: lib/fakeroot.py (hunk 2 above)
Comment (heasterday) [reply lib/fakeroot.py:239]: Can do, I'll just check that it is above version 24.

File: lib/fakeroot.py (hunk 1 above)
Comment (heasterday) [reply lib/fakeroot.py:151]: > Good work. Some comments inline.
> 
> Were you planning to add the Debian 11 (#1043) and openSUSE (#1044) from PR #1035 here?
> 
> See also #1042.

I was going to make an issue for Debian 11 (doesn't appear out yet) and OpenSUSE, looks like you already did which is quite helpful!

Comment (heasterday) [reply lib/fakeroot.py:151]: Will clarify that we install via RPM for both.

File: test/build/50_fakeroot.bats (hunk 4 above)
Comment (heasterday) [reply test/build/50_fakeroot.bats:814]: Let's do `fedora:latest`, I know nothing about how stable their pre-release images are.

File: lib/fakeroot.py [hunk 5]
```diff
@@ -221,6 +224,22 @@
      "cmds": ["apt", "apt-get", "dpkg"],
      "each": ["fakeroot"] },
 
+   # Fedora notes:
+   #
+   # 1. The minimum supported version was chosen somewhat arbitrarily based on the
+   #    release versions available for testing (what was on Dockerhub).
```
Comment (reidpr) [inline lib/fakeroot.py:230]: I'm going to tidy this comment by spelling Docker Hub correctly. You would be correct to point out it's a trivial change; the purpose is to make searches for "Docker Hub" find this spot.

```suggestion
   #    release versions available for testing (what was on Docker Hub).
```

**Review by reidpr — APPROVED** [review]

**Comment by reidpr** [comment]
We decided to put Debian 11 (#1043) and openSUSE (#1044) in a separate PR.

**Pull request merged**